contracts/SpokePool.sol

```diff
@@ -327,19 +338,54 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
             relayAmount: totalRelayAmount
         });
         bytes32 relayHash = _getRelayHash(relayData);
-        _fillRelay(relayHash, relayData, newRelayerFeePct, maxTokensToSend, repaymentChain);
+        uint256 fillAmountPreFees = _fillRelay(relayHash, relayData, newRelayerFeePct, maxTokensToSend, false);
+
+        _emitFillRelay(relayHash, fillAmountPreFees, repaymentChain, relayerFeePct, relayData);
+    }
+
+    function distributeRelaySlow(
+        address depositor,
+        address recipient,
+        address destinationToken,
+        uint64 realizedLpFeePct,
+        uint64 relayerFeePct,
+        uint64 depositId,
+        uint256 originChainId,
+        uint256 totalRelayAmount,
+        uint256 relayerRefundId,
+        bytes32[] memory proof
+    ) public nonReentrant {
+        RelayData memory relayData = RelayData({
+            depositor: depositor,
+            recipient: recipient,
+            destinationToken: destinationToken,
+            realizedLpFeePct: realizedLpFeePct,
+            relayerFeePct: relayerFeePct,
+            depositId: depositId,
+            originChainId: originChainId,
+            relayAmount: totalRelayAmount
+        });
+
+        require(
+            MerkleLib.verifyRelayData(relayerRefunds[relayerRefundId].slowRelayFulfilmentRoot, relayData, proof),
+            "Invalid proof"
+        );
+
+        bytes32 relayHash = _getRelayHash(relayData);
+        uint256 fillAmountPreFees = _fillRelay(relayHash, relayData, relayerFeePct, type(uint256).max / 1e18, true);
```

Comment:
Interesting idea -- what you're saying is that relayAmount >= total send, so this will always result in the relay being completely filled. I like this. Will add an explanatory comment too.